### PR TITLE
Enables FxCop Analyzer for the Connector library

### PR DESCRIPTION
**Pull request by gabog**
Addresses part of #2367 

- Enables FxCop Analyzer for the Connector library and addressed violations.
- Marked OAuthClientOld as obsolete.
- Marked a lot of private methods in AdalAuthenticator as statics (that's the big diff here)

### CODE REVIEW

File: libraries/Microsoft.Bot.Connector/Microsoft.Bot.Connector.csproj [hunk 1]
```diff
@@ -42,6 +42,10 @@
 		<PackageReference Include="Microsoft.IdentityModel.Clients.ActiveDirectory" Version="5.2.4" />
 		<PackageReference Include="Microsoft.IdentityModel.Protocols.OpenIdConnect" Version="5.6.0" />
 		<PackageReference Include="Microsoft.Rest.ClientRuntime" Version="2.3.21" />
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.0.0">
```
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Connector/Microsoft.Bot.Connector.csproj:45]: Indentation

File: libraries/Microsoft.Bot.Connector/OAuthClientConfig.cs [hunk 2]
```diff
@@ -45,92 +47,88 @@ public static async Task SendEmulateOAuthCardsAsync(OAuthClient client, bool emu
...
+                if (shouldTrace)
                 {
-                    // Ignore the exception
+                    ServiceClientTracing.SendRequest(invocationId, httpRequest);
                 }
 
-                ex.Request = new HttpRequestMessageWrapper(httpRequest, requestContent);
-                ex.Response = new HttpResponseMessageWrapper(httpResponse, responseContent);
+                cancellationToken.ThrowIfCancellationRequested();
+                var httpResponse = await client.HttpClient.SendAsync(httpRequest, cancellationToken).ConfigureAwait(false);
```
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Connector/OAuthClientConfig.cs:86]: `HttpResponseMessage` implements IDisposable. Might be a good opportunity to wrap this in a using.

Comment (gabog) [reply libraries/Microsoft.Bot.Connector/OAuthClientConfig.cs]: Done

File: libraries/Microsoft.Bot.Connector/Microsoft.Bot.Connector.csproj (hunk 1 above)
Comment (gabog) [reply libraries/Microsoft.Bot.Connector/Microsoft.Bot.Connector.csproj:45]: Done

**Pull request merged**